in_toto/runlib.py

```diff
@@ -137,18 +138,29 @@ def record_artifacts_as_dict(artifacts):
     original_cwd = os.getcwd()
     try:
       os.chdir(in_toto.settings.ARTIFACT_BASE_PATH)
-    except OSError as e:
-      raise OSError("Review your ARTIFACT_BASE_PATH setting - {}".format(e))
+    except Exception as e:
```

Comment:
Is there a reason why you switched from catching `OSError` to catching `Exception`?  According to the [Python documentation](https://docs.python.org/3/library/os.html#module-os), all functions in the os modules raise `OSError` 

> in the case of invalid or inaccessible file names and paths, or other arguments that have the correct type, but are not accepted by the operating system.

Reply:
It could also be a TypeError. I figured I just catch all Exceptions and re-raise as `SettingsError`.

Reply:
You can also explicitly catch all the exceptions that you're aware of.  For the unexpected exceptions, it might be better to let the exception pass through. 
```
try:
    os.chdir(...)
except (OSError, TypeError) as e:
    ...
```

A problem I have encountered in the past is that the code would sometimes catch weird things like `NameError` and the exception message would be totally wrong.  You append the original exception, so I guess it would be easy to debug / catch these cases.